src/consensus/aft/raft.h

```diff
@@ -1999,30 +2042,41 @@ namespace aft
 
     void add_vote_for_me(const ccf::NodeId& from)
     {
-      size_t quorum = -1;
-
-      if (reconfiguration_type == ReconfigurationType::TWO_TRANSACTION)
+      // Add vote for from node in each configuration where it is present
+      for (auto const& conf : configurations)
       {
-        const auto& cfg = configurations.front();
+        auto const& nodes = conf.nodes;
+        votes_for_me[conf.idx].quorum = get_quorum(nodes.size());
```

Comment:
Why are we calculating this every time?

I think this means we're constructing `votes_for_me` on-the-fly as each vote arrives? I would have expected us to populate this when we become candidate/trigger an election - at _that_ index, we work out and store the quorum in each configuration, and create buckets for the votes to go in. I haven't thought about what it means for configurations to change mid-election.

Since we iterate through configurations here, `votes_for_me` is always big-enough by line 2067, but only just in time.

Reply:
Fixed. We now clear and set `votes_for_me` when becoming candidate or becoming aware of a new term.